### PR TITLE
List VMs by Security Group & HA

### PR DESCRIPTION
**Pull request by GabrielBrascher**
## Description

List VMs by "security group ID" and/or "ha enabled"

Improve List VMs command allowing to list VMs filtering by security group or HA enabled.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

List VMs via API with different combinations. All worked fine.
- filter vms with security group ID
`list virtualmachines filter=id,name,haenable,securitygroup listall=true securitygroupid=<SG UUID>`

- list VMs according to the HA enabled
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=true`
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=false`

- combione both added parameters
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=false securitygroupid=<SG UUID>`
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=true securitygroupid=<SG UUID>`
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=false securitygroupid=<SG UUID2>`
`list virtualmachines filter=id,name,haenable,securitygroup listall=true haenable=true securitygroupid=<SG UUID2>`

### CODE REVIEW

**Comment by GabrielBrascher** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -1016,10 +1024,18 @@
...
 
+        if(securityGroupId != null) {
+            sc.setParameters("securityGroupId", securityGroupId);
+        }
+
         if (display != null) {
             sc.setParameters("display", display);
         }
 
+        if (display != null) {
```
Comment (Pearl1594) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1035]: @GabrielBrascher I guess this is supposed to be isHaEnabled != null ?

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 2]
```diff
@@ -1016,10 +1024,18 @@
             sc.setParameters("serviceOfferingId", serviceOffId);
         }
 
+        if(securityGroupId != null) {
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1027]:
```suggestion
        if (securityGroupId != null) {
```

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java (hunk 1 above)
Comment (sureshanaparti) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1035]: good catch @Pearl1594

Comment (GabrielBrascher) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java]: The code has been updated addressing both comments. Thanks for the review @Pearl1594 and @sureshanaparti.

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by wido** [comment]
LGTM based on code

**Review by sureshanaparti — APPROVED** [review]
Code changes LGTM

**Review by Pearl1594 — COMMENTED** [review]
Tested, LGTM - however, is removal of querying based on affinity group id intentional?

**Comment by wido** [comment]
Very valid comment from @sureshanaparti ! @GabrielBrascher Can you elaborate on that?

**Comment by DaanHoogland** [comment]
Are you talking about @Pearl1594 's comment @wido ? (affinity group affinity ;)

**Comment by wido** [comment]
Indeed @DaanHoogland, I quoted the wrong person. I meant @Pearl1594 his comment indeed.

**Comment by DaanHoogland** [comment]
@GabrielBrascher indeed please advice. As per enough 👍 and test effort I merged but overlooked @Pearl1594 's question. Should we revert or fix forward or is there a legitimate explanation to the answer the query? (cc @wido , @sureshanaparti )

**Comment by sureshanaparti** [comment]
@DaanHoogland seems affinityGroupId code is accidentally removed. As this PR changes impacting existing func, better revert and fix that, and merge it again.

**Comment by GabrielBrascher** [comment]
My mistake @DaanHoogland @Pearl1594 @sureshanaparti. Should we revert this commit?

**Comment by DaanHoogland** [comment]
revert or fix forward @GabrielBrascher your choice for my part,, but with priority please.

**Comment by GabrielBrascher** [comment]
@DaanHoogland @sureshanaparti @Pearl1594 @wido I have created PR #4405 addressing the removed affinity group ID.

I am sorry for all the trouble, guys. This was the first time that I messed this big with a PR.
I really appreciate all the reviews and heads up!

**Pull request merged**